custom_components/waste_collection_schedule/waste_collection_schedule/source/samiljo_se.py

```diff
@@ -125,37 +130,73 @@ def fetch(self):
 
         soup = BeautifulSoup(wasteschedule.text, "html.parser")
 
-        #Calender uses diffrent tags for the last week of the month
-        wastedays = soup.find_all("td", {"style":"styleDayHit"}) + soup.find_all("td", "styleDayHit")
-
         entries = []
-        # get a list of all tags with waste collection days for the current year
-        for wasteday in wastedays:
-            wasteday_wastetype = wasteday.parent.parent
-            # find month and year for given day
-            monthyear = wasteday_wastetype.find_previous(
-                "td", "styleMonthName"
-            ).contents
-            monthyear_parts = str(monthyear).split("-")
-            month = MONTH_MAP[monthyear_parts[0].strip(" []/'")]
-            year = int(monthyear_parts[1].strip(" []/'"))
-
-            #Diffrent tag on collection day
-            day = int(
-                str(wasteday_wastetype.find("div", ["styleInteIdag", "styleIdag"]).contents[0])
-            )
-            # list of bins collected for given day
-            for td in wasteday_wastetype.contents[3].find_all("td"):
-                if td.has_attr("class"):
-                    waste = str(td.get_attribute_list("class")).strip(" []/'")
-                    if waste in NAME_MAP:
-                        t=NAME_MAP.get(waste)
-                        icon=ICON_MAP.get(waste)
-                        entries.append(
-                            Collection(
-                                t=t,
-                                icon=icon,
-                                date=datetime.date(year, month, day),                       
-                            )
-                    )
+        # Extended list of bin types to cover all possible collection types
+        bin_types = [
+            'HKARL1', 'HKARL1-H', 'HKARL2', 'HKARL2-H',
+            'FKARL1', 'FKARL1-H', 'FKARL2', 'FKARL2-H', 
+            'HMAT', 'HMAT-H', 'HREST', 'HREST-H',
+            'HOSORT', 'HOSORT-H', 'FOSORT', 'FOSORT-H',
+            'HREST-HK', 'HREST-HK-H', 'HKARL1-HK', 'HKARL1-HK-H',
+            'TRG', 'TRG-H', 'FREST-HK', 'FREST-HK-H',
+            'FKARL1-HK', 'FKARL1-HK-H', 'FREST', 'FREST-H'
+        ]
+        
+        for bin_type in bin_types:
+            elements = soup.find_all("td", class_=bin_type)
```

Comment:
This seems to be pretty volatile. Is there any other good way to find the entries on the page without the need to hard-code the bin types?